src/main/resources/com/puppycrawl/tools/checkstyle/grammars/javadoc/JavadocParser.g4

```diff
@@ -861,39 +861,41 @@ description: (
 
 reference:
       (
-            PACKAGE (DOT | CLASS)* HASH? MEMBER? parameters?
-            | (DOT | CLASS)+ HASH? MEMBER? parameters?
+            PACKAGE_CLASS HASH? MEMBER? parameters?
             | HASH? MEMBER parameters?
       )
       ;
 parameters: LEFT_BRACE (ARGUMENT | COMMA | WS | NEWLINE | LEADING_ASTERISK)* RIGHT_BRACE;
 
-javadocTag: AUTHOR_LITERAL (WS | NEWLINE)* description?
+javadocTag: AUTHOR_LITERAL (WS | NEWLINE)* ((WS | NEWLINE)+ description)?
 
-    | DEPRECATED_LITERAL (WS | NEWLINE)* description?
+    | DEPRECATED_LITERAL (WS | NEWLINE)* ((WS | NEWLINE)+ description)?
 
-      | EXCEPTION_LITERAL (WS | NEWLINE)* CLASS_NAME? (WS | NEWLINE)* description?
+      | EXCEPTION_LITERAL (WS | NEWLINE)* CLASS_NAME? (WS | NEWLINE)*
+      ((WS | NEWLINE)+ description)?
 
-      | PARAM_LITERAL (WS | NEWLINE)* PARAMETER_NAME? (WS | NEWLINE)* description?
+      | PARAM_LITERAL (WS | NEWLINE)* PARAMETER_NAME? (WS | NEWLINE)*
+      ((WS | NEWLINE)+ description)?
 
-      | RETURN_LITERAL (WS | NEWLINE)* description?
+      | RETURN_LITERAL (WS | NEWLINE)* ((WS | NEWLINE)+ description)?
 
-      | SEE_LITERAL (WS | NEWLINE)* reference? (STRING | htmlElement)* (WS | NEWLINE)* description?
+      | SEE_LITERAL (WS | NEWLINE)* reference? (STRING | htmlElement)*
```

Comment:
Realized a new issue. #4591 . Will add a commit to this PR itself, to speed things up. Would that be alright ?